browser/ui/split_view/split_view_controller.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_SPLIT_VIEW_SPLIT_VIEW_CONTROLLER_H_
+#define BRAVE_BROWSER_UI_SPLIT_VIEW_SPLIT_VIEW_CONTROLLER_H_
+
+#include <optional>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "base/scoped_observation.h"
+#include "brave/browser/ui/split_view/split_view_web_panel_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data_observer.h"
+#include "components/tab_collections/public/tab_interface.h"
+
+class SplitViewBrowserData;
+class SplitViewView;
+class SplitViewWebPanelData;
+class TabStripModel;
+struct TabTile;
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+// Control split view state based on two models(split view tab tile and web
+// panel data).
+// TODO(https://github.com/brave/brave-browser/issues/45475):
+// SplitViewWebPanelData is not implemented yet.
+class SplitViewController : public SplitViewBrowserDataObserver {
+ public:
+  explicit SplitViewController(TabStripModel* tab_strip_model);
+  ~SplitViewController() override;
+
+  SplitViewBrowserData* split_view_browser_data() {
+    return &split_view_tab_tile_data_;
+  }
+
+  void set_split_view_view(SplitViewView* view) { view_ = view; }
```

Comment:
I'm very concerned about lifetime issues here, can you explain how we ensure that `SplitViewController` always lives longer than the view (in comments on the code)

Reply:
`BrowserWindowFeatures` is a member of `Browser` while `SplitView` is a member of `BrowserView` (which owns `Browser`).
```cpp
class Browser {
    std::unique_ptr<BrowserWindowFeatures> features_;
};

class BrowserView : public View /* eventually */ {
    std::unique_ptr<Browser> browser_;
};

class BraveBrowserView : public BrowserView {
    raw_ptr<SplitView> split_view_ = AddChildView()
};
```

Its interesting, so it looks like the `BrowserView` destructor destroys all child views (and nulls out the rawptrs to avoid dangling pointers) so I think this is safe:

dtor order is something like:

```
~BraveBrowserView
~BrowserView
// browser_->GetFeatures().TearDownPreBrowserViewDestruction() called here
~SplitView
~Browser
~BrowserWindowFeatures
~SplitViewController
```